### PR TITLE
[Component] Rollforward of PR#2697

### PR DESCRIPTION
**Pull request by numerology**
Rollforward of #2697 
Provide a default value of mlpipeline ui metadata path.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2760)

### CODE REVIEW

**Comment by numerology** [comment]
/hold cancel
Off line test is passing.

**Comment by numerology** [comment]
/retest
Failure is not due to the bug intended to be fixed.

**Comment by numerology** [comment]
Ok. It seems like the tests are pretty flaky. May need to investigate when we got some cycles.

File: components/gcp/container/component_sdk/python/kfp_component/launcher/__main__.py [hunk 1]
```diff
@@ -26,9 +27,19 @@ def main():
...
+        help='Path for the file where the mlpipeline-ui-metadata.json data '
+             'should be written.')
     parser.add_argument('args', nargs=argparse.REMAINDER)
     args = parser.parse_args()
-    launch(args.file_or_module, args.args)
+
+    if args.ui_metadata_path:
+        os.environ['KFP_UI_METADATA_PATH'] = args.ui_metadata_path
+
+    launch(args.file_or_module, [args.args, args.ui_metadata_path])
```
Comment (Ark-kun) [inline components/gcp/container/component_sdk/python/kfp_component/launcher/__main__.py]: Are you sure the `args.ui_metadata_path` needs to be passed to functions? They usually do no accept it.

Comment (numerology) [reply components/gcp/container/component_sdk/python/kfp_component/launcher/__main__.py]: I've verified it works by building a new container image. But your point is good. The actual root cause might be simply the argument name should be `ui_metadata_path` but not `--ui_metadata_path`.

Let me take a shot to see if that works.

**Pull request merged**